### PR TITLE
Feature/advanced query from file

### PR DESCRIPTION
**Pull request by timabrmsn**
## Description of change

Implements the ability for `alerts search` and `security-data (search|send-to)` commands to read `--advanced-query` arg from a file. 

If the argument string for `--advanced-query` does not begin with an opening brace `{`, the CLI assumes the argument is a filename, and will try to open and parse the file as JSON. Helpful error messages are printed if the file can't be read or the JSON can't be parsed. 

`--advanced-query` is still incompatible with all other query arguments, but it can now be used with a checkpoint.

### CODE REVIEW

File: .github/workflows/cla.yml
Comment (kiran-chaudhary) [inline .github/workflows/cla.yml:1]: This file should not be committed in this branch, its already part of another PR!!

File: CONTRIBUTING.md
Comment (kiran-chaudhary) [inline CONTRIBUTING.md:191]: Part of another PR!!!

**Review by kiran-chaudhary — CHANGES_REQUESTED** [review]

File: .github/workflows/cla.yml
Comment (alanag13) [reply .github/workflows/cla.yml:1]: @timabrmsn this is usually a sign that your working branch is not caught up with master, could you merge master into this?

File: src/code42cli/click_ext/groups.py
Comment (alanag13) [inline src/code42cli/click_ext/groups.py:1]: are all these `click_ext` files the same as they were before, just moved to another directory? What, if anything, has changed here?

File: src/code42cli/cmds/search/options.py
Comment (alanag13) [inline src/code42cli/cmds/search/options.py:126]: `parse` would probably be a better description of what's happening here than `validate`.

File: src/code42cli/cmds/securitydata.py
Comment (alanag13) [inline src/code42cli/cmds/securitydata.py:214]: use `elif` here.

File: src/code42cli/click_ext/types.py
Comment (alanag13) [inline src/code42cli/click_ext/types.py:20]: `curl` does this by treating arguments that have the first character as `@` as files. Could be a good opportunity to borrow a familiar cli pattern.

File: src/code42cli/click_ext/types.py
Comment (alanag13) [inline src/code42cli/click_ext/types.py:22]: these two try/except blocks can merged into a single function that take a string argument (either `value` or file.read()`). It's okay if the exception message is the same for both of those cases, imo.

File: src/code42cli/click_ext/options.py
Comment (alanag13) [inline src/code42cli/click_ext/options.py:1]: has anything changed here?

File: src/code42cli/cmds/alerts.py
Comment (alanag13) [inline src/code42cli/cmds/alerts.py:182]: I would expect to see some sort of comparable/equivalent change made for file events for this, am I missing it somewhere?

File: src/code42cli/click_ext/groups.py
Comment (timabrmsn) [reply src/code42cli/click_ext/groups.py:1]: Yeah, just reorganized things a bit. Since I was adding the new `click.File` subclass (`JSONFileOrString`) and it didn't really fit anywhere, I decided to move all the re-usable custom click classes here.

File: src/code42cli/click_ext/options.py
Comment (unparalleled-js) [reply src/code42cli/click_ext/options.py:1]: Does not look like this - this was in `options.py` before, if that helps.

File: src/code42cli/cmds/securitydata.py
Comment (timabrmsn) [reply src/code42cli/cmds/securitydata.py:214]: Nope, the `saved-search` uses `cls=incompatible_with("advanced_query")`.

File: src/code42cli/cmds/alerts.py
Comment (timabrmsn) [reply src/code42cli/cmds/alerts.py:182]: The change for file events is in the `_call_extractor` function, which is used by both `search` and `send-to`.

File: src/code42cli/cmds/search/options.py
Comment (timabrmsn) [reply src/code42cli/cmds/search/options.py:126]: Well, we've already parsed the json at this point, now we're trying to build the query, so I wanted it to be clear that it's not a json error but that we just couldn't build the query from the data we got. 

Should I just say "failed to build query"?

File: src/code42cli/click_ext/types.py
Comment (timabrmsn) [reply src/code42cli/click_ext/types.py:20]: Yeah, at first I was thinking it would be simpler for the user to just pass a bare filename or query and not make it explicit, but I think it is actually probably better to be.

File: src/code42cli/click_ext/types.py
Comment (timabrmsn) [reply src/code42cli/click_ext/types.py]: I separated the file reading from the json parsing which cleaned this logic up.

File: src/code42cli/cmds/search/options.py
Comment (alanag13) [reply src/code42cli/cmds/search/options.py]: Something like `failed to build query from input JSON` seems to fit that, yeah.

File: CHANGELOG.md
Comment (alanag13) [inline CHANGELOG.md:15]: we should call out the `@` prefix here as well as in the help documentation string.

Comment (timabrmsn) [reply CHANGELOG.md:15]: done

**Review by alanag13 — APPROVED** [review]

**Comment by alanag13** [comment]
@kiran-chaudhary when you get a chance can you re-review/approve this, assuming that the changes you requested have been addressed?

As an FYI, if you request changes, this makes it so that the change cannot be merged until you approve (even if someone else approves).

**Review by kiran-chaudhary — APPROVED** [review]
My bad, apologies. LGTM. I have not tested it, the work has some conflicts with my changes so once merged I will test it along with my work.

**Pull request merged**
